### PR TITLE
Add linting for our test suite as well

### PR DESCRIPTION
**Pull request by SirCmpwn**
And fix all of the linting problems that arise.

### CODE REVIEW

File: src/store/configureStore.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default function configureStore(initialState) {
   if (module.hot) {
     module.hot.accept('../reducers', () =>
       // eslint-disable-next-line global-require
-      store.replaceReducer(require('../reducers').default)
+      store.replaceReducer(require('../reducers')['default'])
```
Comment (eatonphil) [inline src/store/configureStore.js]: What's with this?

File: test/actions/api/linodes.spec.js [hunk 2]
```diff
@@ -25,63 +25,63 @@ describe("actions/linodes/power", sinon.test(() => {
...
+        auth, dispatch, getState, fetchStub,
       }) => {
-      const f = powerOnLinode("foo");
+      const f = powerOnLinode('foo');
 
       await f(dispatch, getState);
 
       expect(fetchStub.calledWith(
-        auth.token, '/linodes/foo/boot', { method: "POST" })).to.be.true;
+        auth.token, '/linodes/foo/boot', { method: 'POST' })).to.equal(true);
```
Comment (eatonphil) [inline test/actions/api/linodes.spec.js]: What's wrong with `to.be.true`?

File: src/store/configureStore.js (hunk 1 above)
Comment (SirCmpwn) [inline src/store/configureStore.js]: One of the linter rules we have is to not use keywords for names of things.

Comment (SirCmpwn) [inline src/store/configureStore.js]: I'm open to removing this linter rule if we feel uncomfortable about it.

File: test/actions/api/linodes.spec.js (hunk 2 above)
Comment (SirCmpwn) [inline test/actions/api/linodes.spec.js]: Two things:

- The linter rule I mentioned in the previous reply
- Using the side effects of properties like this is an antipattern and tbh to.equal(true) is better than to.be.true imo.

File: src/store/configureStore.js (hunk 1 above)
Comment (eatonphil) [inline src/store/configureStore.js]: Ohhh. That makes more sense. It's all good then.

File: test/actions/api/linodes.spec.js (hunk 2 above)
Comment (eatonphil) [inline test/actions/api/linodes.spec.js]: First point: no arguments. Second point: what side effects does this have?

Comment (eatonphil) [inline test/actions/api/linodes.spec.js]: Or what side effects is this exploiting?

Comment (SirCmpwn) [inline test/actions/api/linodes.spec.js]: Invoking .to.be.true will evaluate the true property of the fluent object, which uses a getter to assert the value is true and raise an exception (this is the side effect) if it's not.

Comment (SirCmpwn) [inline test/actions/api/linodes.spec.js]: In other words, 'foo.bar` should just give you a value, and instead you could use `foo.bar()` if you need to produce side effects.

Comment (eatonphil) [inline test/actions/api/linodes.spec.js]: Got it.

**Comment by eatonphil** [comment]
Looks good. Ready to merge?

**Comment by SirCmpwn** [comment]
Yep.

**Pull request merged**